crates/ruma-events/src/call/member/member_state_key.rs

```diff
@@ -16,39 +16,32 @@ pub struct CallMemberStateKey {
 
 impl CallMemberStateKey {
     /// Constructs a new CallMemberStateKey there are three possible formats:
-    /// - `_{UserId}_{DeviceId}` example: `_@test:user.org_DEVICE`. `device_id: Some`, `underscore:
-    ///   true`
-    /// - `{UserId}_{DeviceId}` example: `@test:user.org_DEVICE`. `device_id: Some`, `underscore:
-    ///   false`
-    /// - `{UserId}` example: `@test:user.org`. `device_id: None`, underscore is ignored:
+    /// - `_{UserId}_{MemberId}` example: `_@test:user.org_DEVICE_m.call`. `member_id:
+    ///   Some(DEVICE_m.call)`, `underscore: true`
+    /// - `{UserId}_{MemberId}` example: `@test:user.org_DEVICE_m.call`. `member_id:
+    ///   Some(DEVICE_m.call)`, `underscore: false`
+    /// - `{UserId}` example: `@test:user.org`. `member_id: None`, underscore is ignored:
     ///   `underscore: false|true`
     ///
-    /// Dependent on the parameters the correct CallMemberStateKey will be constructed.
-    pub fn new(user_id: OwnedUserId, device_id: Option<OwnedDeviceId>, underscore: bool) -> Self {
-        CallMemberStateKeyEnum::new(user_id, device_id, underscore).into()
+    /// The MemberId is a combination of the DeviceId, the UserId and the session
+    /// (session.application and session.id) and the user id. Dependent on the parameters the
+    /// correct CallMemberStateKey will be constructed.
+    pub fn new(user_id: OwnedUserId, member_id: Option<String>, underscore: bool) -> Self {
+        CallMemberStateKeyEnum::new(user_id, member_id, underscore).into()
     }
 
     /// Returns the user id in this state key.
     /// (This is a cheap operations. The id is already type checked on initialization. And does
     /// only returns a reference to an existing OwnedUserId.)
+    ///
+    /// It is recommended to not use the state key to get the user id, but rather use sender field.
```

Comment:
If it is not recommended to use this, maybe we should not have the function?

If we keep it:
```suggestion
    /// It is recommended to not use the state key to get the user id, but rather use the `sender` field.
```

Reply:
In that case we could drop the whole type. Even though it is not recommended this is the enforced type by the unstable specification. So it is correct for the sdk to be specific about this field because everything else would be not to spec.

The way owned state will work in the future however is still in the air so the comment is mostly here to protect users of the sdk from unnecassary maintainance work in case. of a spec change.

Reply:
Okay, could you apply my suggestion then?